src/main/java/dk/cngroup/trainings/spring/springassignment/service/user/UserServiceImpl.java

```diff
@@ -1,24 +1,32 @@
 package dk.cngroup.trainings.spring.springassignment.service.user;
 
-import dk.cngroup.trainings.spring.springassignment.exception.UserEmailExistsException;
+import dk.cngroup.trainings.spring.springassignment.exception.user.UserEmailExistsException;
+import dk.cngroup.trainings.spring.springassignment.model.MyUserDetails;
 import dk.cngroup.trainings.spring.springassignment.model.User;
 import dk.cngroup.trainings.spring.springassignment.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
-public class UserServiceImpl implements UserService {
+public class UserServiceImpl implements UserService, UserDetailsService {
 
 	private UserRepository userRepository;
+	private BCryptPasswordEncoder bCryptPasswordEncoder;
 
 	public UserServiceImpl(UserRepository userRepository) {
 		this.userRepository = userRepository;
+		this.bCryptPasswordEncoder = new BCryptPasswordEncoder();
```

Comment:
you already have this defined as a Bean, so maybe it should be autowired?

Reply:
I think that was before we spoke the @SpringBootTest
With Autowired and no SpringBootTest, some tests fails.
I will fix it.